### PR TITLE
feat: reject mixed system scope

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:
* Reject system scope mixed with user or patient scope 
* Update unit tests
* Remove dependency from combo test to smartHandlerTest file to avoid duplicative tests run

### CODE REVIEW

File: src/smartScopeHelper.test.ts [hunk 1]
```diff
@@ -561,3 +562,19 @@ describe('convertScopeToSmartScope', () => {
...
+        expect(() => {
+            rejectInvalidScopeCombination(['system/Patient.read', 'user/Organization.read']);
+        }).toThrowError(UnauthorizedError);
+    });
+    test('reject system scope mixed with patient scope', () => {
+        expect(() => {
+            rejectInvalidScopeCombination(['system/Patient.read', 'patient/Observation.read']);
+        }).toThrowError(UnauthorizedError);
+    });
+    test('allow patient scope mixed with user scope', () => {
```
Comment (rsmayda) [inline src/smartScopeHelper.test.ts:577]: Could you add a test for a single system scope and a multiple system scopes

File: src/smartScopeHelper.ts [hunk 2]
```diff
@@ -145,6 +145,17 @@ export function isScopeSufficient(
     return false;
 }
 
+export function rejectInvalidScopeCombination(scopes: string[]): void {
+    if (
+        scopes.some((scope: string) => scope.startsWith('system')) &&
+        (scopes.some((scope: string) => scope.startsWith('user')) ||
+            scopes.some((scope: string) => scope.startsWith('patient')))
```
Comment (rsmayda) [inline src/smartScopeHelper.ts]:
```suggestion
        scopes.some((scope: string) => scope.startsWith('system/')) &&
        (scopes.some((scope: string) => scope.startsWith('user/')) ||
            scopes.some((scope: string) => scope.startsWith('patient/')))
```

File: src/smartScopeHelper.ts [hunk 3]
```diff
@@ -145,6 +145,17 @@ export function isScopeSufficient(
...
+    ) {
+        console.log(scopes);
```
Comment (rsmayda) [inline src/smartScopeHelper.ts]: Are we using a logger in FWoA?
`const logger = getComponentLogger();`

File: src/smartHandler.ts [hunk 4]
```diff
@@ -124,6 +125,7 @@ export class SMARTHandler implements Authorization {
 
         // get just the scopes that apply to this request
         const scopes = getScopes(decodedToken[this.config.scopeKey]);
+        rejectInvalidScopeCombination(scopes);
         const usableScopes = filterOutUnusableScope(
```
Comment (rsmayda) [inline src/smartHandler.ts:129]: on this line but on line 156 could you change `return scope.startsWith('system');` to `return scope.startsWith('system/');`

File: src/smartScopeHelper.ts (hunk 2 above)
Comment (rsmayda) [reply src/smartScopeHelper.ts]: Better yet lets not do this but make use of Regex (safer option), can we make this change across all 'startsWith' usage?

Comment (Bingjiling) [reply src/smartScopeHelper.ts]: Sure, I can make that change.

File: src/smartScopeHelper.ts (hunk 3 above)
Comment (Bingjiling) [reply src/smartScopeHelper.ts]: Oops, this was added for debugging. Will remove.

**Review by rsmayda — APPROVED** [review]
Please get second review

**Review by ssvegaraju — APPROVED** [review]
LGTM!

**Pull request merged**